src/Context/Wishlist/Wishlist-Context.jsx

```diff
@@ -0,0 +1,31 @@
+import React, { createContext, useContext, useState } from "react";
+import { Products } from "../../components";
+
+const WishlistContext = createContext();
+
+const WishlistProvider = ({ children }) => {
+  const [wishlist, setWishlist] = useState([]);
+
+  const wishlistToggleHandler = (product) => {
+    // console.log(product);
+    // wishlist.findIndex((item) => item._id == product._id) !== -1
+    //   ? setWishlist([...wishlist, product])
+    //   : wishlist.filter((i) => i._id !== product._id);
+    setWishlist((prevList) => {
+      const index = prevList.findIndex((item) => item._id === product._id);
+      return index === -1
+        ? [...prevList, product]
+        : prevList.filter((p) => p._id !== product._id);
```

Comment:
suggestion: This can be written like this
` const index = prevList.findIndex((item) => item._id === product._id)=== -1;
      return index 
        ? [...prevList, product]
        : prevList.filter((p) => p._id !== product._id);`